overlord/hookstate/ctlcmd/is_connected.go

```diff
@@ -0,0 +1,97 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/overlord/ifacestate"
+)
+
+type isConnectedCommand struct {
+	baseCommand
+
+	Positional struct {
+		PlugOrSlotSpec string `positional-args:"true" positional-arg-name:"<plug|slot>"`
+	} `positional-args:"true" required:"true"`
+}
+
+var shortIsConnectedHelp = i18n.G(`Return success if the given plug or slot is connected, and failure otherwise`)
+var longIsConnectedHelp = i18n.G(`
+The is-connected command returns success if the given plug or slot of the
+calling snap is connected, and failure otherwise.
+
+$ snapctl is-connected plug
```

Comment:
what's the actual output here? afaiu there will be some on error

Reply:
Indeed, good point, there is an error message. I've updated the help string and spread test.

Reply:
It's actually:
`error: error running snapctl: network is not connected`

Not sure if we want to do something about it (other than me updating the doc).

Reply:
Two considerations:

* we want to remove the "error running snapctl" wrapping in daemon in general in think, in most cases people will see those errors originating from "snapctl" anyway
* I'm not quite sure what the output should exactly be but we want to gain a bit more control for when errors are expected in snapctl: let's introduce in ctlcmd

type UnsuccesfulError {
    ExitCode int
}

then let's use the kind mechanism of API results so that the client can detect it, the behavior of cmd/snapctl when it gets an api error for that is to still print stdout and stderr and then os.Exit with the UnsuccessfulError exit code. Instead of just existing immediately after printing "error: ..."

Reply:
@stolowski  better a follow up at this point, but we still want to remove the "error running snapctl:" wrapping, don't know if that affects snapctl spread tests etc

Reply:
Will do.